analyses/mutational-signatures/07-plot_cns_fit.Rmd

```diff
@@ -0,0 +1,485 @@
+  
+```
+
+
+
+<br><br>
+
+### Explore: MMR2 signature and mutation burden
+
+Which samples have MMR ~ 1, and are these hypermutators? 
+This analysis is discussed [here](https://github.com/AlexsLemonade/OpenPBTA-analysis/pull/1227#issuecomment-1064493652).
+
+Note that this visualiation is strictly exploratory and not exported. 
+
+We do **not** observe that samples with high MMR2 exposure have high mutational burdens. We do see that the majority of all such samples are subtyped as HGG, H3 wildtype.
+
+```{r mmr2, fig.width = 10, fig.height = 3}
+top_mmr2 <- results_df %>%
+  dplyr::filter(signature == "MMR2") %>%
+  dplyr::select(exposure, Kids_First_Biospecimen_ID, cancer_group_display) %>%
+  dplyr::distinct() %>%
+  dplyr::arrange(-exposure) %>%
+  dplyr::inner_join(tmb_data) %>%
+  tidyr::gather(dplyr::contains("tmb"), key = "tmb_type", value = "tmb")
+
+
+# We see in this visualization that very few samples are >0.5 MMR2 exposure,
+# and that the samples with extremely high mutation counts are NOT the same as those with high MMR2 exposure
+top_mmr2 %>%
+  ggplot2::ggplot() + 
+  ggplot2::aes(x = exposure, y = tmb, fill = cancer_group_display) +
+  ggplot2::geom_point(shape = 21, color ="black") + 
+  ggsci::scale_fill_simpsons() +
+  ggplot2::labs(x = "Exposure to MMR2") +
+  ggplot2::facet_wrap(~tmb_type, scales = "free_y")
+
+# What are the specific samples that are > MMR2 exposure 0.75? (Since so few samples are this high, we can view this more below ~1.)
+top_mmr2 %>%
+  dplyr::filter(exposure >= 0.75) %>%
+  dplyr::select(-dplyr::contains("tmb")) %>%
+  dplyr::distinct() %>%
+  dplyr::inner_join(
+    meta %>%
+      dplyr::select(Kids_First_Biospecimen_ID, molecular_subtype)
+  ) %>%
+  knitr::kable()
+
+
+
+```
+
+##### Explore: High TMB signatures
+
+When exploring MMR2, we see that the four extremely mutated samples are _not_ dominantly exposed to MMR2. What are these samples, and what signatures are they exposed to?
+
+All four samples are High-grade glioma astrocytoma with subtype	`HGG, H3 wildtype`, and two are tp53 loss while the other two are tp53 activated.
+All of these samples have _no exposure_ to signatures 3, 8, or N6. Two of the samples (one tp53 loss and other tp53 activated) have no exposure to signature 19. 
+
+```{r hypermut}
+# Grab the IDs for the four samples
+hypermut_ids <- tmb_data %>%
+  dplyr::filter(tmb > 2) %>%
+  dplyr::pull(Kids_First_Biospecimen_ID)
+
+
+# What are their subtypes?
+subtypes <- meta %>%
+  dplyr::filter(Kids_First_Biospecimen_ID %in% hypermut_ids) %>%
+  dplyr::select(Kids_First_Biospecimen_ID, molecular_subtype) 
+
+# Table of merged information about these four samples:
+results_df %>% 
+  dplyr::filter(Kids_First_Biospecimen_ID %in% hypermut_ids) %>%
+  dplyr::select(Kids_First_Biospecimen_ID, cancer_group_display, exposure, signature) %>%
+  dplyr::inner_join(subtypes) %>%
+  dplyr::group_by(Kids_First_Biospecimen_ID) %>%
+  tidyr::spread(signature, exposure) %>%
+  knitr::kable()
```

Comment:
I do not find this table to be particularly easy to read. I think it might be how the signature numbers are aligned and the inconsistent width. 

This is a later problem _but_ it's kind of difficult to tell if these samples show a consistent pattern that's _any different_ from all other tumors in that subtype. I could imagine an exposures stacked bar plot (within HGG, H3 wildtype) that's faceted by hypermutated vs. not.

Reply:
If/when we do this, I'd maybe recommend splitting everything in the explore section into its own notebook just to keep things more digestible. You'd presumably want to save some things from earlier in the notebook in tabular form.

Reply:
Noting I removed the pipe into `knitr::kable()`.